\documentclass[a4paper,12pt]{article}
\usepackage{graphicx}
\usepackage{hyperref}
\usepackage[all]{xypic}
\usepackage{amsmath,amssymb}

\title{Identifying loci under selective pressure using a PPP-value classifier}
\author{Toby Dylan Hocking}

\begin{document}

\maketitle

\newcommand{\fig}[3][1]{
  \begin{figure}[htp]
    \begin{center}          %linewidth
    \includegraphics[width=#1\textwidth]{#2}
    \end{center}
    \caption{#3\label{#2}}
  \end{figure}
}
% \brat = brace matrix
\newcommand{\brat}[2]{
  \left[
    \begin{array}{#1}
      #2
    \end{array}
    \right]
}
\newcommand{\neff}{\text{popsize}}
\newcommand{\aij}{\alpha_{ij}(t)}
\newcommand{\aijs}{\alpha_{ij}^*(t)}
\newcommand{\wij}[1]{w_{ij}^{\text{#1}}}
\newcommand{\etal}{\emph{et al.}}
\newcommand{\RR}{\mathbb R}
\newcommand{\Bin}{\operatorname{Binomial}}

\section{Introduction}

The explosion of molecular marker data in animal populations from
technologies such as Single Nucleotide Polymorphism (SNP) microarrays
opens new avenues of research for population genetics. These data
allow testing of many aspects of our current models of population
genetics, such as the evolutionary status of these markers relative to
selective pressures. One area we are interested in is the estimation
of genetic differentiation between populations, and establishing
methods for determining which markers and genomic regions have been
under selective pressure. Genomic areas under selection are areas with
probable functional significance, thus the goal is to develop methods
we can use to identify functional genes and augment the current state
of functional annotations for domestic animal genomes.

In the last several decades, the statistical tools of biologists and
geneticists have evolved considerably. In particular, modern computers
and stochastic methods such as Markov Chain Monte Carlo (MCMC) allow
for complex Bayesian models of evolutionary systems. In this work, we
investigate one such Bayesian model used to describe evolution of
domestic animation populations, and extend it with a classifier for
markers under selection.

The model of domestic animal evolution that we consider in this work
is the hierarchical Bayesian model of Nicholson \etal
(2002)\cite{nicholson}, hereafter referred to as the Nicholson
model. This model attempts to estimate ancestral population allele
frequencies for each marker, and a measure of divergence for each
population which is analogous to the classical $F_{st}$ of population
genetics.

The new idea put forth in this work is a classifier for markers under
selection, based on Posterior Predictive P-values
(PPP-values)\cite{pppvalues}. Essentially these PPP-values are the
Bayesian analogue of the usual frequentist P-values, which indicate
departures from the model hypotheses. The Nicholson model was designed
for, and accurately models, independently evolution populations under
genetic drift. However, loci under selection in addition to genetic
drift represent departures from the model hypotheses. Thus we use
PPP-values estimates from the model to identify these aberrent loci.

Furthermore, we validate the model using a simulation of evolution by
genetic drift and selection. The simulator has been implemented using
the R programming language\cite{R}. The model fitting has been
implemented using compiled FORTRAN code dynamically linked to R. The
simulations, analyses, and graphics discussed in this article can be
reproduced by using the code published in the R package
\texttt{nicholsonppp} on R-Forge\cite{R-Forge}:

 \url{http://nicholsonppp.r-forge.r-project.org/}

\section{Methods}

\subsection{Simulating genetic drift and selection}

To simulate selection and genetic drift in several independent
populations, we use a modified version of the simulator in
\cite{Beaumont-Balding}. We assume $L$ independent loci, and $P$
independent populations, evolving over $T$ generations. To model SNP
data, each locus has only 2 possible alleles, thus we denote the
allele frequency for locus $i$ in population $j$ at generation $t$ as
$\aij$, with $0\leq \aij\leq 1$. To assign ancestral allele
frequencies $\pi_i=\alpha_{1,j}(1)=...=\alpha_{L,j}(1)$, we draw from a truncated
$\beta(0.7,0.7)$ distribution \autoref{beta}.

\fig[0.5]{beta}{Distribution of ancestral allele frequencies in our
  simulations follow a truncated $\beta(0.7,0.7)$ distribution.}

If we assign the colors blue and red to the 2 alleles, and we define
$\aij$ is the blue allele frequency, then $1-\aij$ is the red allele
frequency. The color of the allele will determine its fate under
selection in populations which each have a background color of either
red or blue. Population color is chosen at random, independently for
each locus, at the beginning of the simulation, with probability 0.4
for each of red or blue, and 0.2 for neutral populations where neither
allele is favored. The idea is to simulate the fact that some alleles
are favored in some environments, while disfavored in others. Thus,
under positive selection, red alleles will be favored in red
populations, and disfavored in blue populations (vice versa for blue
alleles). Under balancing selection, it is advantageous to have both a
blue and red allele. This simulates the heterozygote advantage, a
phenomenon classically observed in the gene that controls malaria
resistence and sickle-cell anemia affectation.

The allele frequency changes in each generation via
2 mechanisms: drift and selection. Drift introduces some random
variability up or down in allele frequency, independent of population
color:
$$\aijs = \operatorname{rbinom}(\neff,\alpha_{ij}(t-1))$$
The effect of drift grows more important relative to selection as
population size $\neff$ diminishes.

Then to update the allele frequency for selection, we first calculate
relative fitness of each diploid genotype. Relative fitness of a locus
is based on the selection coefficient for that locus $s_i\in \RR$,
which is a parameter of the simulation, usually between 0 and 1 in
empirical studies. Selection for a locus grows more important relative
to genetic drift as $s_i$ increases.
$$
\begin{array}{cccll}
\wij{BB} & \wij{BR} & \wij{RR} & \text{selection type} & \text{population color}\\
\hline
1 & 1+s_i/2 & 1+s_i & \text{positive}& \text{red}\\
1+s_i & 1+s_i/2 & 1 & \text{positive}& \text{blue}\\
1 & 1+s_i & 1 & \text{balancing}& \\
1 & 1 & 1 & \text{neutral}& 
\end{array}
$$
Then we update blue allele frequency for selection based on
Hardy-Weinberg equilibrium:
$$\aij = \frac{
\wij{BB}\aijs^2 + \wij{BR}\aijs[1-\aijs]/2 }{
\wij{BB}\aijs^2 + \wij{BR}\aijs[1-\aijs] + \wij{RR}[1-\aijs]^2
   }$$

   We repeat the process for $t=2, ..., T$, and we take values
   $\alpha_{ij}(T)$ as the output allele frequencies of the
   simulation.



\subsection{The hierarchical bayesian Nicholson model}

To model the variation between observed allele frequencies in
different populations, the Nicholson model assigns a divergence
parameter $c_j$ to each population. The number of observed alleles is
modeled as $x_{ij}\sim \Bin(\neff,\alpha_{ij})$, where $\alpha_{ij}$
is the population allele frequency for locus $i$ and population
$j$. This quantity is in turned modeled by $\alpha_{ij}\sim N(\pi_i,
c_j \pi_i(1-\pi_i))$, a normal distribution truncated to the interval
[0,1]. The differentiation parameter $c_j$ is motivated by population
genetics (see section 2.2, \cite{nicholson}). The distributions of the
hyperparameter for ancestral allele frequency follows $\pi_i\sim
\beta(a,a)$ and the population divergence parameter follows $c_j\sim
U[0,1]$.

The relationship between model parameters is more clearly summarised
in the following diagram:

$$
\xymatrix{
  \pi_i \ar[rd] & & c_j \ar[ld] \\
  & \alpha_{ij} \ar[d] & \\
  & (X_{ij},N_{ij})
}
$$

MCMC techniques are used to sample from the following posterior distributions:
$$\alpha^t = P(\alpha|c^{t-1},\pi^{t-1},x)$$
$$\pi^t = P(\pi|c^{t-1},\alpha^t,a)$$
$$c^t = P(c|\pi^t,\alpha^t)$$

\subsection{PPP-value calculation}

Calculation of PPP-values in the Markov Chain is used for the
classifier. For each iteration $t$ through the chain, we say the data fits
the model if the following criterion is met:

$$
p_i^t =
\begin{cases}
1 & \text{if }\left(
\frac{\operatorname{rbinom}(N_{ij},\alpha_{ij}^t)}{N_{ij}} - \pi_i^t
\right)^2
>
\left(
\frac{Y_{ij}}{N_{ij}} - \pi_i^t
\right)^2\\
0 & \text{otherwise}
$$

Thus the PPP-value for locus $i$ is given by

$$
p_i = \frac 1 T \sum_{t=1}^T p_i^t
$$

\section{Results}


Loci fixation?

 Some alleles can be fixed at the end of the simulation,
  $\aij\in\{0,1\}$
 Since we are modeling Single Nucleotide Polymorphisms the data
  are probably not fixed.
 We can consider non-fixed loci only, for the model fitting
  later.
 Criteria:
\begin{description}
\item[not.all.fixed] Throw out the locus if all subpopulations fixed.
\item[none.fixed] Throw out the locus if one or more subpopulations fixed.
\end{description}


\subsection{Simulation verification}

\fig{loci-over-time}{Allele frequency evolution of 6 loci in 12
  populations over 100 generations.}

\fig[0.5]{fixation-selection}{foo}

\fig{sim-summary-plot}{foo}

To visualize the influence of the number of generations on each of
these diagnostic plots, we used to the animation
package\cite{animation} to create a series of plots, one for each
generation. These images are put together and viewed in sequence to
form a statistical animation that reveals the dependence on the number
of generations. The animations can be viewed on the accompanying
website:

 \url{http://nicholsonppp.r-forge.r-project.org/}

\subsection{Model estimates}

\fig{anc-est-plot}{To diagnose dependence of model fit on selection
  state, we plot ancestral allele frequency estimates for each loci
  versus actual values from the simulation. Neutral loci are well
  estimated by the model, but loci under balancing and positive
  selection are not well estimated.}

\fig{2009-08-27-notbeta}{Scatter plot matrix for various values of
  ancestral allele frequency $\pi$ for a simulated data set (actual
  simulated values indicated by row/column simulated). Models using
  priors that follow a $\beta(1,1)$ (indicated by fortranr1 and
  winbugs1) and $\beta(0.7,0.7)$ (fortran.old, fortranr0.7, and
  winbugs0.7) were fit using WinBUGS and our FORTRAN program. For
  alleles under positive selection, there are small discrepancies
  between the FORTRAN and WinBUGS programs.}

\fig[0.5]{c-over-time-all}{foo} \fig[0.5]{c-over-time-diff}{foo}
\fig[0.5]{c-over-time-same}{foo} \fig[0.8]{cutoff-plot-few}{foo}

\subsection{Sensitivity of the PPP-value classifier}

\fig[0.8]{dens-several-s-few}{foo} 
\fig[0.8]{dens-several-s-neu}{foo}

\fig[0.8]{cutoff-plot-neu}{foo}
\fig[0.8]{cutoff-plot}{foo}
\fig[0.8]{dens-several-s}{foo}
\fig{fixation-endpoints}{foo}
\fig{gen-pop-dens}{foo}
\fig{gen-pop}{foo}
%\fig{gen-pop-roc}{foo}
%\fig{roc-desc}{foo}
%\fig[0.8]{roc-s}{foo}

\section{Conclusions}

\end{document}
